src/utils/objectUtils.ts

```diff
@@ -94,6 +96,30 @@ export function groupBy<
   )
 }
 
+/**
+ * @param array The array of objects to be grouped.
+ * @param selector The key used for grouping the objects. Support nested keys.
+ * @returns An object where the keys are unique values from the given selector and the values are the corresponding objects from the array.
+ */
+export function groupByPath<T extends object>(array: T[], selector: string): Record<string, T[]> {
```

Comment:
🟢 How would it behave with nested arrays of objects? I am not saying we need to cover that case, I am just curious about whether it throws an error or not

Reply:
I can create a test, but it is not intended to be used that way. It will return empty `Record = {}`

Reply:
Do you mention this scenario?
```ts
      const input: TestType1[] = [
        {
          id: 1,
          name: 'a',
          bool: true,
          nested: [{ code: 100 }],
        },
      ]
```
'nested.code'

Reply:
Yes, exactly, I know it is not intended to be used in that way, I was just curious to know what to expect :D thanks for sharing :)